pkg/kritis/cryptolib/verify_detached_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import (
+	"encoding/base64"
+	"testing"
+	//   "crypto"
+	// 	 "crypto/sha256"
+	// 	 "encoding/base64"
+	// 	 "crypto/rand"
+	// 	 "crypto/rsa"
+	// 	 "encoding/pem"
+	// 	 "crypto/x509"
+	// 	 "fmt"
+)
+
+const good_plaintext = "good payload"
+
+// RSA Public Keys
+const rsa_2048_pub_key = `-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA1Vf+g3iALHB5GHz2sTMm
+VmemVMio8iAgIXDhfdJmAJJCUG61vH7ZFOJNWwVPiX2HE4iC8quLT3a2W5h81OiD
+Y2FXfD5vZB0lycXNZoasyhUlC4TFsL01tgh7W7WN5iBDlwxY13bcgv79SIbroz/C
++kS1+cqu4GQXmEHLYFg80pQVe7ssBaQ3qxA0HL0heXJfM0Ye40Aw3aC430h92f2a
+5JgY8JEqRTtYgh3VVuqzm3L4QvSWiHzfB29BXCB0GstFK448aYkk2RPI6Q1LkoT7
+NiCquPVF1EYnUXKrC+ANoWr/l7ldEJ7V4vpo+9EClnXcxAzq/knqeN5WdM6iPYny
+qwIDAQAB
+-----END PUBLIC KEY-----`
+const rsa_4096_pub_key = `-----BEGIN PUBLIC KEY-----
+MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAte+CWFUwOwyD/VBaK55o
+qxTMOX34Y3evVlCY9iDXwYbi+yDgSgjB4K2sTvyV6ztQEBWzReNX3GG7WkPWkHVB
+OIssRgtVUtNnngQkgoOwJWxGf0k1R/NrXRsRMReyuW1zT6iftVhClexf5eBLtmqy
+7hXCpy7NNx9486YpF8KJGa1wrV3Ko3R1qHYjGgKd/fjtQ5CcFUZUYgmYQeIkeDO1
+DiQOSYS2xnkUTfV2aHIgFUlrxdU2M50He5Z/jSE49/9IXQac599y2m6irVaVT6VM
+vY6mxSRGgRx3hkoajVrxd3MglIFLO8/FfrOGwZg9bZFIlgHjUiFPPKbn8bSyldMZ
+xNj36t7YC2ZpN6XGzeBnnLfBW1GjGomk3G1eyBUCwRk9nHt9Dsqr/xEauMocHcfA
+EfEm8/fwcp5MB8GOZ3l8p9Rs38yoBFzVbj2SeicFoQTnHe4DrtMvBzGyKsiaKszV
+5UJAPyqm7zPph/Ck++NRQDmJvkhLXcs7QrUG2TBA7veqaLzrU88jY3RDKSIbNtLA
+xBrmQID1ms7TVrCcB6nWrxHJc3FhTP6q/lUL0ziG8fOND9oW6P0YxaXj+1qIxn45
+x2j5oDwsx8ig4/V23Yuofvji2j/VYAY/KAkKJOXqfozbmlMNUCpN6HuQ5yakVFgW
+XanFz99rw4kZ0z70ziCBp68CAwEAAQ==
+-----END PUBLIC KEY-----`
+
+// Signatures created with RSA Keys. Signatures created with rsa.SignPKCS1v15 and rsa.SignPSS methods from crypto/rsa package.
```

Comment:
Instead of one comment, add a comment saying created by rsa.SignBlah(variables, you, used)  for each of these signatures. Same for ecdsa signatures

When writing crypto tests, use comments to make it as easy as possible for someone with minimal knowledge to fix up test cases if they need to change one part of of it.

Reply:
done. Some of these use the same function call the only difference being the private key. I did not include the private keys in this file. I figure if someone needs to change a signature they can do so with their own key pair and update the public key here.

It may be worth keeping the public and private keys in a testdata directory and read those in the test files.